SpringProject/src/main/java/com/gradproject2019/users/service/UserServiceImpl.java

```diff
@@ -64,6 +64,14 @@ public UserResponseDto editUser(UUID token, Long userId, UserPatchRequestDto use
         return new UserResponseDto().from(getUserById(userId));
     }
 
+    @Override
+    public List<UserResponseDto> findUserByManagerId(UUID token, Long managerId) {
+        checkTokenMatchesUser(token, managerId);
```

Comment:
Not quite: it is absolutely necessary in your framework to have a dao that does just this filter, but what you want here is a general `getUsersOp` that can choose and control the right database query to run. 

So for an intial cut of this, you might have a line of code that checks managerId is not null and is a positive integer, and if it is run the filter users by manager id, if not return an unfiltered list of users. 

Subquequently you'll want to choose the filters based on iterating the query params as a map, and include paging - _but not right now_!